File: Leetcode/Easy/34-Squares-of-a-Sorted-Array/solution.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left,right = 0,len(nums)-1
        res = []
        while left <= right:
            if nums[left]*nums[left] > nums[right]*nums[right]:
                res.append(nums[left]*nums[left])
                left += 1
            else:
                res.append(nums[right]*nums[right])
                right -= 1
        return res[::-1]
                

File: Leetcode/Easy/34-Squares-of-a-Sorted-Array/solution.ts
function sortedSquaress(nums: number[]): number[] {
    let [left,right] = [0,nums.length-1];
    let res:number[] = [];
    while(left <= right){
        if(nums[left]*nums[left] > nums[right]*nums[right]){
            res.push(nums[left]*nums[left]);
            left++;
        }else{
            res.push(nums[right]*nums[right]);
            right--;
        }
    }
    return res.reverse();
};